src/bunit.web/EventDispatchExtensions/TriggerEventDispatchExtensions.cs

```diff
@@ -45,16 +45,47 @@ public static Task TriggerEventAsync(this IElement element, string eventName, Ev
 
 		var isNonBubblingEvent = NonBubblingEvents.Contains(eventName.ToLowerInvariant());
 
+		var unwrappedElement = element.Unwrap();
 		if (isNonBubblingEvent)
-			return TriggerNonBubblingEventAsync(renderer, element.Unwrap(), eventName, eventArgs);
+			return TriggerNonBubblingEventAsync(renderer, unwrappedElement, eventName, eventArgs);
 
-		return TriggerBubblingEventAsync(renderer, element.Unwrap(), eventName, eventArgs);
+		return unwrappedElement switch
+		{
+			IHtmlInputElement { Type: "submit", Form: not null } input when eventName is "onclick" =>
+				TriggerFormSubmitBubblingEventAsync(renderer, input, eventArgs, input.Form),
+			IHtmlButtonElement { Type: "submit", Form: not null } button when eventName is "onclick" =>
+				TriggerFormSubmitBubblingEventAsync(renderer, button, eventArgs, button.Form),
+			_ => TriggerBubblingEventAsync(renderer, unwrappedElement, eventName, eventArgs)
+		};
+	}
+
+	private static Task TriggerFormSubmitBubblingEventAsync(ITestRenderer renderer, IElement element, EventArgs
+	eventArgs, IHtmlFormElement form)
+	{
+
```

Comment:
```suggestion
	private static Task TriggerFormSubmitBubblingEventAsync(ITestRenderer renderer, IElement element, EventArgs eventArgs, IHtmlFormElement form)
	{
```

Just a few whitespace nit picks.